tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -4528,14 +4523,55 @@ Status ConvertFusedBatchNorm(OpConverterParams* params) {
     }
   }
 
-  nvinfer1::ScaleMode mode = nvinfer1::ScaleMode::kCHANNEL;
-  nvinfer1::IScaleLayer* layer = params->converter->network()->addScale(
-      *tensor->trt_tensor(), mode, combined_offset_weights->GetTrtWeights(),
-      combined_scale_weights->GetTrtWeights(),
-      nvinfer1::Weights{nvinfer1::DataType::kFLOAT, nullptr, 0});
-  TFTRT_RETURN_ERROR_IF_NULLPTR(layer, node_def.name());
-  params->converter->SetLayerName(layer, node_def);
-  ITensorProxyPtr output_tensor = layer->getOutput(0);
+  ITensorProxyPtr output_tensor;
+
+  if (data_format == "NCHW") {
+    nvinfer1::ScaleMode mode = nvinfer1::ScaleMode::kCHANNEL;
+    nvinfer1::IScaleLayer* layer = params->converter->network()->addScale(
+        *tensor->trt_tensor(), mode, combined_offset_weights->GetTrtWeights(),
+        combined_scale_weights->GetTrtWeights(),
+        nvinfer1::Weights{nvinfer1::DataType::kFLOAT, nullptr, 0});
+    TFTRT_RETURN_ERROR_IF_NULLPTR(layer, node_def.name());
+    params->converter->SetLayerName(layer, node_def);
+    output_tensor = layer->getOutput(0);
+  }
+  if (data_format == "NHWC") {
+    nvinfer1::Dims dims = tensor->getDimensions();
+    for (int i = 0; i < dims.nbDims - 1; i++) {
+      dims.d[i] = 1;
+    }
+    dims.d[dims.nbDims - 1] = nweight;
+
+    nvinfer1::IConstantLayer* combined_scale_constant_layer =
+        params->converter->network()->addConstant(dims,
+            combined_scale_weights->GetTrtWeights());
+    TFTRT_RETURN_ERROR_IF_NULLPTR(combined_scale_constant_layer, node_def.name());
+    params->converter->SetLayerName(combined_scale_constant_layer, node_def, "scale");
+    ITensorProxyPtr combined_scale_tensor =
+        combined_scale_constant_layer->getOutput(0);
+
+    nvinfer1::IConstantLayer* combined_offset_constant_layer =
+        params->converter->network()->addConstant(dims,
+            combined_offset_weights->GetTrtWeights());
+    TFTRT_RETURN_ERROR_IF_NULLPTR(combined_offset_constant_layer, node_def.name());
+    params->converter->SetLayerName(combined_offset_constant_layer, node_def, "offset");
+    ITensorProxyPtr combined_offset_tensor =
+        combined_offset_constant_layer->getOutput(0);
+
+    nvinfer1::IElementWiseLayer* layer1 = params->converter->network()->addElementWise(
+        *tensor->trt_tensor(), *combined_scale_tensor->trt_tensor(),
+        nvinfer1::ElementWiseOperation::kPROD);
+    TFTRT_RETURN_ERROR_IF_NULLPTR(layer1, node_def.name());
+    params->converter->SetLayerName(layer1, node_def, "prod");
+    ITensorProxyPtr prod_tensor = layer1->getOutput(0);
+    nvinfer1::IElementWiseLayer* layer2 = params->converter->network()->addElementWise(
+        *prod_tensor->trt_tensor(), *combined_offset_tensor->trt_tensor(),
+        nvinfer1::ElementWiseOperation::kSUM);
+    TFTRT_RETURN_ERROR_IF_NULLPTR(layer2, node_def.name());
+    params->converter->SetLayerName(layer2, node_def, "sum");
+    output_tensor = layer2->getOutput(0);
```

Comment:
Consider using [TRTNetworkBuilder](https://github.com/tensorflow/tensorflow/blob/4dbddc4034bdb225dbe7e7efd46e987d39eedff0/tensorflow/compiler/tf2tensorrt/convert/ops/layer_utils.h#L41) helper class, it has methods `Add`, `Mul` and `WeightsToConstant`.

Reply:
Cool, didn't know that existed. Updated the const and eltwise layers to use that now instead.